toplevel/coqloop.ml

```diff
@@ -337,11 +336,15 @@ let loop_flush_all () =
 
 let rec loop () =
   Sys.catch_break true;
-  if !Flags.print_emacs then Vernacentries.qed_display_script := false;
-  Flags.coqtop_ui := true;
   try
     reset_input_buffer stdin top_buffer;
-    while true do do_vernac(); loop_flush_all () done
+    (* Tail-rec is essential *)
+    let rec loop sid =
```

Comment:
I will change the name, but sorry, the loop there is a real loop and IMVHO we should avoid using references gratuitously.